### PR TITLE
CI: Add ci for VS/msvc build

### PR DESCRIPTION
**Pull request by qnfm**
This add ci test for code generated by msvc. 

The full detail is [here](https://github.com/open-quantum-safe/oqs-provider/pull/256#issuecomment-1727750617)

It captured a new error by the test msvc_native:
```
test 4
    Start 4: oqs_tlssig

4: Test command: D:\dev\src\oqs-provider\_build\test\Debug\oqs_test_tlssig.exe "oqsprovider" "D:/dev/src/oqs-provider/test/openssl-ca.cnf" "D:/dev/src/oqs-provider/_build/test/tmp"
4: Working Directory: D:/dev/src/oqs-provider/_build
4: Environment variables:
4:  OPENSSL_MODULES=D:/dev/src/oqs-provider/_build/lib
4:  OPENSSL_CONF=D:/dev/src/oqs-provider/test/openssl-ca.cnf
4: Test timeout computed to be: 10000000
4: OPENSSL_Uplink(00007FFB1067CFE8,08): no OPENSSL_Applink
4/5 Test #4: oqs_tlssig .......................***Failed    0.02 sec
OPENSSL_Uplink(00007FFB1067CFE8,08): no OPENSSL_Applink

test 5
    Start 5: oqs_endecode

5: Test command: D:\dev\src\oqs-provider\_build\test\Debug\oqs_test_endecode.exe "oqsprovider" "D:/dev/src/oqs-provider/test/openssl-ca.cnf"
5: Working Directory: D:/dev/src/oqs-provider/_build/test
5: Environment variables:
5:  OPENSSL_MODULES=D:/dev/src/oqs-provider/_build/lib
5:  OPENSSL_CONF=D:/dev/src/oqs-provider/test/openssl-ca.cnf
5: Test timeout computed to be: 10000000
5: OPENSSL_Uplink(00007FFB1067CFE8,08): no OPENSSL_Applink
5/5 Test #5: oqs_endecode .....................***Failed    0.02 sec
OPENSSL_Uplink(00007FFB1067CFE8,08): no OPENSSL_Applink


0% tests passed, 5 tests failed out of 5
```

All the tests have the same error

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -199,3 +199,117 @@ jobs:
...
+          if: steps.cache-openssl32.outputs.cache-hit != 'true'
+          with:
+            platform: ${{ matrix.platform.arch }}
+        - name: Setup perl for OpenSSL build
+          uses: shogo82148/actions-setup-perl@v1
+          if: steps.cache-openssl32.outputs.cache-hit != 'true'
+# OQS_USE_OPENSSL=OFF by default on Win32
+# if cmake --build fails, try explicit
+# cd _build && msbuild ALL_BUILD.vcxproj -p:Configuration=Release
+# fails:          cmake -DCMAKE_C_FLAGS="/wd5105" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_INSTALL_PREFIX="c:\liboqs" ${{ matrix.platform.oqsconfig }} -S . -B _build
```
Comment (baentsch) [inline .github/workflows/windows.yml:258]: Now, which of these commands can be used? It seems, nothing is executed in this branch (?)

Comment (baentsch) [reply .github/workflows/windows.yml:258]: Or is this just a prior "build liboqs" trial (that probably should be removed)?

**Review by baentsch — CHANGES_REQUESTED** [review]
Thanks for adding this MSVC CI! Besides the separate comment nit, it should pass before merge, so looking at things without much Windows knowledge:

> All the tests have the same error

googling found [this](https://github.com/confluentinc/confluent-kafka-dotnet/issues/1708#issuecomment-998689196): Worthwhile (necessary?) adding this "applink.c" to the test programs? It feels pretty weird having to do this for something testing a DLL, but then again, I don't "get" Windows....

**Comment by baentsch** [comment]
What also makes me wonder: Why does MSVC-2019 pass and 2022 fail?

**Comment by qnfm** [comment]
> What also makes me wonder: Why does MSVC-2019 pass and 2022 fail?

That's the output of ci building liboqs for Windows 2019 with Visual Studio 16 2019
```
"D:\a\oqs-provider\oqs-provider\liboqs\build\INSTALL.vcxproj" (default target) (1) ->
"D:\a\oqs-provider\oqs-provider\liboqs\build\ALL_BUILD.vcxproj" (default target) (3) ->
"D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj" (default target) (4) ->
(ClCompile target) -> 
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(487,17): error C2059: syntax error: '/' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(502,17): error C2059: syntax error: '/' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(530,17): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(531,13): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(533,9): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(534,5): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(665,16): error C2079: 'varDefaultValue' uses undefined struct 'tagVARIANT' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(950,16): error C2079: 'varValue' uses undefined struct 'tagVARIANT' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(319,24): error C2059: syntax error: '/' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(378,37): error C2371: 'pvarVal': redefinition; different basic types [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(379,9): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(380,5): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(383,3): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(384,1): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(487,17): error C2059: syntax error: '/' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(502,17): error C2059: syntax error: '/' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(530,17): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(531,13): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(533,9): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(534,5): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(665,16): error C2079: 'varDefaultValue' uses undefined struct 'tagVARIANT' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(950,16): error C2079: 'varValue' uses undefined struct 'tagVARIANT' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(319,24): error C2059: syntax error: '/' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(378,37): error C2371: 'pvarVal': redefinition; different basic types [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(379,9): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(380,5): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(383,3): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\propidlbase.h(384,1): error C2059: syntax error: '}' [D:\a\oqs-provider\oqs-provider\liboqs\build\src\common\common.vcxproj]
```

In this [post](https://stackoverflow.com/questions/15734367/error-c2059-syntax-error-when-adding-c-source-files-to-project), it may arise from the lack support of C99 initialization style (designated initializers) in VS 16 2017.
As shown in [table](https://learn.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-160#c-standard-library-features-1): The [Designated initialization](https://wg21.link/p0329r4) is supported in VS 2019 16.1.

**Comment by qnfm** [comment]
Updated:
The liboqs build error was [fixed ](https://github.com/qnfm/liboqs/blob/ccebcfae2b82b2e08c3f4d0cffdb7748cb1b421c/.CMake/compiler_opts.cmake)by enabling C11 standard option in cmake files.
```
add_compile_options(/std:c11)
```

File: .github/workflows/windows.yml (hunk 1 above)
Comment (qnfm) [reply .github/workflows/windows.yml:258]: deleted

**Pull request closed without merging**